src/types/resolveDescriptors.ts

```diff
@@ -264,6 +265,7 @@ export function resolveDescriptors(ctx: CompilerContext) {
     const types: Map<string, TypeDescription> = new Map();
     const staticFunctions: Map<string, FunctionDescription> = new Map();
     const staticConstants: Map<string, ConstantDescription> = new Map();
+    const structDependencies: Map<string, Set<TypeDescription>> = new Map();
```

Comment:
Can you please add a comment on what "struct dependencies" are?

Reply:
This is a Map that stores Structs on which contracts and other Structs depend

Reply:
do you mean for each contract we get the set of structs that can be actually instantiated in it and not just declared?

if so, what happens if a contract does not ever call a static function that touches a struct? or what if a contract refers statically to a struct but we can never tell if it is ever going to be created at runtime?

what I'm saying is something like: do we actually need to put that struct information into the ABI file?

Reply:
Hm, good point. There can be cases when struct is used only in runtime and never in sent/received messages - and in such cases there's no sense to put it into ABI.

I guess it'll be right to only analyse message structs used contract either as receiver selector or in message sending.

Reply:
why don't we remove structs from abi altogether?

Reply:
> or in message sending

this might happen in unreachable code

Reply:
> why don't we remove structs from abi altogether?

but they are part of the "interface" of the contract. this is how to interact with it 🤔

Reply:
> > or in message sending
> 
> this might happen in unreachable code

error throwing could also happen in unreachable code

Reply:
I believe it's better to have some struct/error in ABI even if it's not actually being used there in real scenarios, than not having them at all.

The original problem was that structs/errors that go into ABI were selected in a wrong way so that **the same contract** in different compiling projects would generate a **different** ABI.